calculation/spatial_hazard/spatial_hazard/correlate_ims.py

```diff
@@ -78,150 +77,85 @@ def calculate_distance_matrix(
     return pd.DataFrame(index=stations, data=distance_matrix, columns=stations)
 
 
-def get_corr_matrix(stations, distance_matrix, selected_im, emp_df):
+def get_corr_matrix(
+    stations: Sequence[str], distance_matrix: pd.DataFrame, selected_im: gc.im.IM
+):
     """
     Computes the correlation matrix for the specified stations
     using the Loth & Baker (2013) model.
 
     Parameters
     ----------
-    stations: sequence of strings
-    distance_matrix: dataframe
+    stations: Sequence[str]
+        List of the station names
+    distance_matrix: pd.DataFrame
+        Distance values between all site locations
     selected_im: IM
-    emp_df: dataframe
-
-    Returns
-    -------
-    R: dataframe
-        The correlation matrix of the given station
+        IM to get correlations from the model
     """
-    assert np.all(
-        distance_matrix.index.values == emp_df.index.values
-    ), "Order of the stations has to be the same"
-
-    # Compute correlation matrix
     R = np.eye(len(stations))
-    for i, station_i in enumerate(stations):
-        correlation = models.loth_baker_model.get_correlations(
+    for i, station in enumerate(stations):
+        correlation = sha_calc.spatial_hazard.loth_baker_model.get_correlations(
             str(selected_im),
             str(selected_im),
-            distance_matrix.loc[stations, station_i].values,
+            distance_matrix.loc[stations, station].values,
         )
         assert np.all(
             correlation >= 0.0
-        ), "Correlation should be positive or 0. Error with {}, {}".format(i, station_i)
+        ), f"Correlation should be positive or 0. Error with {i}, {station}"
         R[i, :] = R[:, i] = correlation
 
-    R = pd.DataFrame(index=stations, columns=stations, data=R)
-    return R
+    return pd.DataFrame(index=stations, columns=stations, data=R)
 
 
 def create_random_fields(
```

Comment:
I think this can be renamed to something `gen_IM_values` or similar, makes it a bit clearer what it does imo